performance_regression_gradual_grow_throughput.py

```
@@ -602,6 +645,123 @@ def run_gradual_increase_load(self, workload: Workload, stress_num, num_loaders,
+        For each node in parallel:
+        1. Create the dump directory on the remote node.
+        2. Call the REST stop endpoint which stops profiling AND writes per-shard
+           folded-stack files to the given path.
+        3. Verify at least one dump file exists and is non-empty.
+        4. Create a tar.gz archive on the remote node.
+        5. Download the archive into node.logdir/ via receive_files.
+        6. Clean up remote files.
+        """
+
+        def stop_and_collect_on_node(node):
+            dump_dir = f"/tmp/async_profile_{step_name}_{node.name}"
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/async_profile_"

(S108)